src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ThrottlingRerouteSpec.groovy

```diff
@@ -50,55 +59,54 @@ class ThrottlingRerouteSpec extends HealthCheckSpecification {
-            breakFlow(it, false)
-            //don't sleep here, since there is already an antiFlapMin delay between actual port downs
-        }
+        flowPaths.collect { breakFlow(it, false) }
+        // Don't sleep here, since there is already an antiFlapMin delay between actual port downs
         def rerouteTriggersEnd = new Date().time
         /*At this point all reroute triggers have happened. Save this time in order to calculate when the actual
         reroutes will happen (time triggers stopped + reroute delay seconds)*/
 
         then: "The oldest broken flow is still not rerouted before rerouteDelay run out"
         Wrappers.wait(rerouteDelay * 3) {
-            assert flowHelper.getLatestHistoryEntry(flows.first().flowId).action == "Flow rerouting"
+            assert flows.first().retrieveFlowHistory().entries.last().action == "Flow rerouting"
             // wait till reroute starts
         }
-        def rerouteTimestamp = flowHelper.getLatestHistoryEntry(flows.first().flowId).timestampIso
+        def rerouteTimestamp = flows.first().retrieveFlowHistory().entries.last().timestampIso
         // check time diff between the time when reroute was triggered and the first action of reroute in history
         def differenceInMillis = flowHelper.convertStringTimestampIsoToLong(rerouteTimestamp) - rerouteTriggersEnd
         // reroute starts not earlier than the expected reroute delay
         assert differenceInMillis > (rerouteDelay) * 1000
         // reroute starts not later than 2 seconds later than the expected delay
-        assert differenceInMillis < (rerouteDelay + 2) * 1000
+        assert differenceInMillis < (rerouteDelay + 2.5) * 1000
 
         and: "The oldest broken flow is rerouted when the rerouteDelay runs out"
         def untilReroutesBegin = { rerouteTriggersEnd + rerouteDelay * 1000 - new Date().time }
         def waitTime = untilReroutesBegin() / 1000.0 + PATH_INSTALLATION_TIME * 2
         Wrappers.wait(waitTime) {
-            //Flow should go DOWN or change path on reroute. In our case it doesn't matter which of these happen.
-            assert (northboundV2.getFlowStatus(flows.first().flowId).status == FlowState.DOWN &&
-                    flowHelper.getHistoryEntriesByAction(flows.first().flowId, REROUTE_ACTION).find {
-                        it.taskId =~ (/.+ : retry #1/) })||
-                    northbound.getFlowPath(flows.first().flowId) != flowPaths.first()
+            // Flow should go DOWN or change path on reroute.
+            // In our case it doesn't matter which of these happen.
+            def flow1 = flows.first()
+            assert (flow1.retrieveFlowStatus().status == FlowState.DOWN &&
+                    flow1.retrieveFlowHistory().entries.findAll {it.getAction() == REROUTE_ACTION}
+                            .find { it.taskId =~ (/.+ : retry #1/) })||
+                    flow1.retrieveAllEntityPaths().getMainPath() != flowPaths.first()
```

Comment:
let's use `flowPaths.find{ it.flowId == flow1.flowId}` instead of` flowPaths.first()`
(flowPath.flowId has been added in the scope of PR 5673)

Reply:
Done

Reply:
upd: I've took the `flowId: flowPathPayload.id`  from your PR 5673, so now `flowPath.flowId` is not `null`